### PR TITLE
Adding validation output for testing results

### PR DESCRIPTION
**Pull request by hpractv**

### CODE REVIEW

**Comment by hpractv** [comment]
Here's what the test results look like for a doc with warnings and failures:
image

File: README.md [hunk 1]
```diff
@@ -18,6 +19,12 @@ This action is used to validate a YAML file with a custom [schema](#schema-file)
 | `yaml-file-path`   | true        | The path of the yaml file to validate.                                                                               |
 | `schema-file-path` | true        | The schema file used to validate yaml file.  If omitted or set to "SAM", the IM-OPEN SAM schema format will be used. |
 
+## Outputs
+
+| Parameter            | Description                                                                     |
+| -------------------- | ------------------------------------------------------------------------------- |
+| `validation-outcome` | The results of the file validation. Will container success, warning, or failed. |
```
Comment (danielle-casella-adams) [inline README.md:26]:
```suggestion
| `validation-outcome` | The results of the file validation. Will be success, warning, or failed. |
```

File: README.md [hunk 2]
```diff
@@ -28,10 +35,15 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Test SAM YAML
-        uses: im-open/yaml-file-validation@v1.0.0
+        uses: im-open/yaml-file-validation@v1.0.1
```
Comment (danielle-casella-adams) [inline README.md:38]:
```suggestion
        id: sam-test
        uses: im-open/yaml-file-validation@v1.0.1
```

File: action.yml [hunk 3]
```diff
@@ -8,6 +9,12 @@ inputs:
     description: 'The schema file used to validate yaml file.  If omitted or set to ""SAM"", the IM-OPEN SAM schema format will be used.'
     required: true
     default: 'SAM'
+
+outputs:
+  validation-outcome:
+    description: 'The results of the file validation. Will container success, warning, or failed.'
```
Comment (danielle-casella-adams) [inline action.yml:15]:
```suggestion
    description: 'The results of the file validation. Will be success, warning, or failed.'
```

File: dist/index.js [hunk 4]
```diff
@@ -3932,19 +3940,25 @@ try {
...
+        warn('Document #' + docNumber + ' has warnings.');
       } else {
         info('Document #' + docNumber + ' successfully validated.');
       }
       info('Finished Validating Document #' + docNumber);
       docNumber++;
       docFailed = false;
+      docWarn = false;
     });
+    core.setOutput('validation-outcome', hasFailure ? 'failed' : hasWarn ? 'warning' : 'success');
```
Comment (danielle-casella-adams) [inline dist/index.js:3961]: You can keep it like this but the double ternary is hard to read.  Would it be possible to not do a final core.setOutput and just put a core.setOutput in each of the sections above?

**Review by danielle-casella-adams — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (hpractv) [reply README.md]: 👍🏽

File: README.md (hunk 2 above)
Comment (hpractv) [reply README.md:39]: Missed some of my copy/paste stuff!

File: action.yml (hunk 3 above)
Comment (hpractv) [reply action.yml]: 👍🏽

File: dist/index.js (hunk 4 above)
Comment (hpractv) [reply dist/index.js:3961]: The sam.yaml files can have multiple documents. That makes me think there should be an overall result.

**Review by danielle-casella-adams — DISMISSED** [review]

**Review by JosephDSchwartz — APPROVED** [review]

**Pull request merged**